packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,359 @@
+    - [Multiple File selection:](#multiple-file-selection)
+  - [Controlled Usage](#controlled-usage)
+    - [Direct Upload with Progress Display](#direct-upload-with-progress-display)
+- [Accessibility](#accessibility)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=78150%3A2625&mode=design&t=bKQvPInRmAoeAAFV-1) to all variants of the `FileUpload` component.
+
+## Anatomy
+
+<img width="100%" src="./file-upload-anatomy.png" alt="Single File Upload">
+<img width="100%" src="./file-upload-item-anatomy.png" alt="Multiple Files Upload">
+
+## `FileUpload` Props
+
+```ts
+// Check the File type from MDN for more details: https://developer.mozilla.org/en-US/docs/Web/API/File
+interface BladeFile extends File {
+  /**
+   * The unique identifier of the file.
+   */
+  id: string;
+  /**
+   * The file's upload status.
+   */
+  status?: 'uploading' | 'success' | 'error';
+  /**
+   * The percentage of file upload completion.
+   */
+  percent?: number;
+  /**
+   * Text indicating an error state
+   */
+  errorText?: string;
+}
+
+type BladeFileList = BladeFile[];
+
+type FileUploadProps = {
+  /**
+   * Label for the FileUpload component
+   */
+  label?: string;
+  /**
+   * Accessibility label for the FileUpload component, required if label is not provided
+   */
+  accessibilityLabel?: string;
+  /**
+   * Position of the label relative to the file upload area. Desktop only prop. Default value on mobile will be 'top'
+   */
+  labelPosition?: 'top' | 'left';
+  /**
+   * Defines the selection behavior of the FileUpload component
+   */
+  selectionType?: 'single' | 'multiple';
+  /**
+   * File types that can be accepted. See [input's accept attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept)
+   *
+   * Usage: accept=".jpg, .png, .pdf", accept="image/*", accept="image/png, image/jpeg, application/pdf"
+   */
+  accept?: string;
+  /**
+   * Disables or enables the FileUpload component
+   */
+  isDisabled?: boolean;
+  /**
+   * The name of the file upload input, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)
+   */
+  name?: string;
+  /**
+   * Default list of files that have been uploaded, useful when the component is uncontrolled
+   */
+  defaultFileList?: BladeFileList;
+  /**
+   * List of files that have been selected/uploaded, useful when the component is controlled
+   */
+  fileList?: BladeFileList;
+  /**
+   * Limit the number of files that can be uploaded
+   */
+  maxCount?: number;
+  /**
+   * Limit the size of the uploaded files (in bytes)
+   */
+  maxSize?: number;
+  /**
+   * Callback function triggered when files are selected
+   */
+  onChange?: ({ name, fileList }: { name: string; fileList: BladeFileList }) => void;
+  /**
+   * Callback function triggered when the preview icon is clicked
+   */
+  onPreview?: (previewedFile: BladeFile) => void;
```

Comment:
what happens when preview is clicked? like why is it clickable?

Reply:
@saurabhdaware For example, you uploaded a file and now you want to check if you have uploaded/selected the correct file you can click on the preview button to open & see it.

For example, onPreview can be used in the following after the image has been uploaded:

https://github.com/razorpay/blade/assets/46647141/31162e42-5347-4468-b493-5187d85dcacf


There are two ways how we can approach it:

- Don't show the preview icon by default, show it only when `onPreview` is provided. 
- Always show the preview icon by providing a default behavior of opening the file in a new tab which is overridden by the `onPreview` prop.

Reply:
I can add this as an open question too

Reply:
Thinking more on this, we can't predict the default behavior it could be different for different file types. So, I'm inclined on the 1st behavior mentioned above.